examples/platform/silabs/efr32/rs911x/hal/efx32_ncp_host.c

```diff
@@ -0,0 +1,506 @@
+#define SL_WIFI_BOARD_READY_WAIT_TIME \
+  40000 //some scenarios like after firmware upgrade, it will take 40 seconds to boad ready
+#endif
+
+typedef struct {
+  sl_wifi_buffer_t *head;
+  sl_wifi_buffer_t *tail;
+  osMutexId_t mutex;
+  uint32_t flag;
+} si91x_packet_queue_t;
+
+osEventFlagsId_t si91x_events   = 0;
+osEventFlagsId_t si91x_bus_events   = 0;
+osEventFlagsId_t si91x_async_events = 0;
+osThreadId_t si91x_thread       = 0;
+osMutexId_t si91x_bus_mutex     = 0;
+osThreadId_t si91x_event_thread = 0;
+
+static si91x_packet_queue_t cmd_queues[SI91X_QUEUE_MAX];
+
+extern void si91x_bus_thread(void *args);
+extern void si91x_event_handler_thread(void *args);
+static void gpio_interrupt(uint8_t interrupt_number);
+sl_status_t sli_verify_device_boot(uint32_t *rom_version);
+sl_status_t sli_wifi_select_option(const uint8_t configuration);
+
+void sl_si91x_host_set_sleep_indicator(void)
+{
+  GPIO_PinOutSet(WFX_SLEEP_CONFIRM_PIN.port, WFX_SLEEP_CONFIRM_PIN.pin);
+}
+
+void sl_si91x_host_clear_sleep_indicator(void)
+{
+  GPIO_PinOutClear(WFX_SLEEP_CONFIRM_PIN.port, WFX_SLEEP_CONFIRM_PIN.pin);
+}
+
+uint32_t sl_si91x_host_get_wake_indicator(void)
+{
+  return GPIO_PinInGet(WAKE_INDICATOR_PIN.port, WAKE_INDICATOR_PIN.pin);
+}
+
+sl_status_t sl_si91x_host_init(void)
+{
+  int i;
+  sl_status_t status = SL_STATUS_OK;
+
+  status = sl_si91x_host_bus_init();
+  VERIFY_STATUS(status);
+
+    // configure packet pending interrupt priority
+  NVIC_SetPriority(GPIO_ODD_IRQn, PACKET_PENDING_INT_PRI);
+#if 0
+  // Start reset line low
+  GPIO_PinModeSet(WFX_RESET_PIN.port, WFX_RESET_PIN.pin, gpioModePushPull, 0);
+
+  // configure packet pending interrupt priority
+  NVIC_SetPriority(GPIO_ODD_IRQn, PACKET_PENDING_INT_PRI);
+
+  // Configure interrupt, sleep and wake confirmation pins
+  GPIOINT_CallbackRegister(WFX_INTERRUPT_PIN.pin, rsi_gpio_irq_cb);
+  GPIO_PinModeSet(WFX_INTERRUPT_PIN.port, WFX_INTERRUPT_PIN.pin, gpioModeInputPullFilter, 0);
+  GPIO_ExtIntConfig(WFX_INTERRUPT_PIN.port, WFX_INTERRUPT_PIN.pin, SL_WFX_HOST_PINOUT_SPI_IRQ, true, false, true);
+  GPIO_PinModeSet(WFX_SLEEP_CONFIRM_PIN.port, WFX_SLEEP_CONFIRM_PIN.pin, gpioModeWiredOrPullDown, 1);
+  GPIO_PinModeSet(WAKE_INDICATOR_PIN.port, WAKE_INDICATOR_PIN.pin, gpioModeWiredOrPullDown, 0);
+#endif
+  memset(cmd_queues, 0, sizeof(cmd_queues));
+
+  if (NULL == si91x_events) {
+    si91x_events = osEventFlagsNew(NULL);
+  }
+  if (NULL == si91x_bus_events) {
+    si91x_bus_events = osEventFlagsNew(NULL);
+  }
+  if (NULL == si91x_async_events) {
+    si91x_async_events = osEventFlagsNew(NULL);
+  }
+
+  if (NULL == si91x_thread) {
+    const osThreadAttr_t attr = {
+
+      .name       = "si91x_bus",
+      .priority   = osPriorityRealtime,
+      .stack_mem  = 0,
+      .stack_size = 1536,
+      .cb_mem     = 0,
+      .cb_size    = 0,
+      .attr_bits  = 0u,
+      .tz_module  = 0u,
+    };
+    si91x_thread = osThreadNew(si91x_bus_thread, NULL, &attr);

```

Comment:
they are not allowing CMSIS API for now .

All other code is still using FreeRTOS API's, change it.